types/minio/minio-tests.ts

```diff
@@ -370,3 +370,7 @@ minio.extensions.listObjectsV2WithMetadata('testBucket');
 minio.extensions.listObjectsV2WithMetadata('testBucket', 'test_');
 minio.extensions.listObjectsV2WithMetadata('testBucket', 'test_', true);
 minio.extensions.listObjectsV2WithMetadata('testBucket', 'test_', true, 'some_object.jpg');
+
+// @ts-expect-error
```

Comment:
I believe it's necessary to get it working without error suppression. As I understand, there should be a possibility to call `setRequestOptions` without params. In this case we need to change the declaration like that (adding question mark):
```typescript
setRequestOptions(options?: RequestOptions): void;
```

- [ ] Sort it out

Reply:
I think @MysteryBlokHed 's code is correct. 

According to [minio.js](https://github.com/minio/minio-js/blob/20de703aa4ff84c4440d3759e87170904c91e707/src/main/minio.js#L205), setRequestOptions function is called with param `option`.  If parameter is `null`, it throws TypeError. (check out [isObject](https://github.com/minio/minio-js/blob/4fce54bd81188aaa86fe6927786877cd71c2b79b/src/main/helpers.js#L233) code)
```js
setRequestOptions(options) {
    if (!isObject(options)) {
      throw new TypeError('request options should be of type "object"')
    }
    this.reqOptions = _.pick(options, ['agent', 'ca', 'cert', 'ciphers', 'clientCertEngine', 'crl', 'dhparam', 'ecdhCurve', 'family', 'honorCipherOrder', 'key', 'passphrase', 'pfx', 'rejectUnauthorized', 'secureOptions', 'secureProtocol', 'servername', 'sessionIdContext'])
  }
```

Any other opinion? 😃

Reply:
Agreed; Since the only purpose of the function is to change options, it wouldn’t make sense for the options parameter to be empty

Reply:
I didn't get why we need to write wrong method call which produces an error? It's the only place in the whole file when we use error suppression. If we remove those lines what will be changed? It seems like we can just remove them.

Is not that matter anyway, let's merge it

```typescript
// @ts-expect-error
minio.setRequestOptions();
```